### PR TITLE
RHOAIENG-11274: chore(Makefile): widen the container build context to the whole repo, so we can start reusing code between images in the future

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-11274

## Description

```python
import glob
import pathlib

def main():
    for dockerfile in glob.glob("**/Dockerfile", recursive=True):
        print(dockerfile)
        path = pathlib.Path(dockerfile)
        lines = path.read_text().splitlines()
        idx = lines.index("ARG SOURCE_CODE=.")
        lines[idx] = f"ARG SOURCE_CODE={str(path.parent)}"
        print("\t", lines[idx])
        path.write_text("\n".join(lines) + "\n")

if __name__ == '__main__':
    main()
```

TODOs

* [ ] the next change is to manually edit the SOURCE_CODE variable to point to shared locations for workbench builds that all build from the same files that now need to be duplicated to each directory https://issues.redhat.com/browse/RHOAIENG-16518 is a pilot task for jupyterlab loading spinner
* [ ] this will break the change detection mechanism in GHA for PRs; we can't no longer so easily tell what files are used by what Dockerfile https://issues.redhat.com/browse/RHOAIENG-16517
* [ ] previous point is solvable if the GHA reads the Dockerfile and checks what it accesses in COPY and ADD instructions
* [x] update openshift/release, https://github.com/openshift/release/pull/59520

## How Has This Been Tested?

GHA on my fork

* https://github.com/jiridanek/notebooks/actions/runs/12140010357

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .dockerignore
[the remark this answers is not in the document]
Comment (jiridanek) [reply .dockerignore:1]: /retest

[the remark this answers is not in the document]
Comment (jiridanek) [reply .dockerignore:1]: /testwith jiridanek/release/jd_change_context/notebook-codeserver-ubi9-python-3-11-pr-image-mirror

[the remark this answers is not in the document]
Comment (jiridanek) [reply .dockerignore:1]: /testwith jiridanek/release/jd_change_context/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

[the remark this answers is not in the document]
Comment (jiridanek) [reply .dockerignore:1]: /testwith openshift/release/master/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

[the remark this answers is not in the document]
Comment (jiridanek) [reply .dockerignore:1]: /testwith openshift/release/master/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith jiridanek/release/jd_change_context/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/notebook-codeserver-ubi9-python-3-11-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/ci/prow/runtime-rocm-tensorflow-ubi9-python-3-9-pr-image-mirror openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/opendatahub-io/images openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/images openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/ci/images openshift/release#59520 opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/ci/prow/images openshift/release#59520 opendatahub-io/notebooks#792

**Comment by atheo89** [comment]
/testwith opendatahub-io/notebooks/main/ci/prow/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror jiridanek/release#jd_change_context

**Comment by atheo89** [comment]
/testwith opendatahub-io/notebooks/main/ci/prow/runtime-minimal-ubi9-python-3-9-pr-image-mirror openshift/release#59520

**Comment by atheo89** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror openshift/release#59520

**Comment by atheo89** [comment]
/testwith opendatahub-io/notebooks/main/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror openshift/release#59520

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror openshift/release#59520
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror openshift/release#59520

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror openshift/release#59520
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror openshift/release#59520

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror openshift/release#59520

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-11-pr-image-mirror openshift/release#59520

**Comment by caponetto** [comment]
/lgtm with https://github.com/openshift/release/pull/59520

Are we tracking the test failures?
```
- cuda-jupyter-tensorflow-ubi9-python-3.11
AttributeError: 'Sequential' object has no attribute 'output_names'

- rocm-jupyter-pytorch-ubi9-python-3.9
AssertionError: '2.4' != '2.2'
- 2.4
?   ^
+ 2.2
?   ^
 : incorrect version

- rocm-jupyter-tensorflow-ubi9-python-3.11
AssertionError: '2.14' != '2.17'
- 2.14
?    ^
+ 2.17
?    ^
 : incorrect version
```

**Comment by jiridanek** [comment]
> Are we tracking the test failures?

We mention it on every retrospective, OCP-CI notebooks tests never passing, ... no Jira that I know about

**Comment by jiridanek** [comment]
pushed it back to what has been lgtmd

image

**Comment by jiridanek** [comment]
/override "build (cuda-jupyter-tensorflow-ubi9-python-3.11) / build"
/override "build (rocm-jupyter-pytorch-ubi9-python-3.9) / build"
/override "build (rocm-jupyter-tensorflow-ubi9-python-3.11) / build"

Failures tracked in https://issues.redhat.com/browse/RHOAIENG-16587

**Comment by jiridanek** [comment]
/override multi-pr-opendatahub-io-notebooks-792-openshift-release-59520-notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror

**Comment by jiridanek** [comment]
/override multi-pr-opendatahub-io-notebooks-792-openshift-release-59520-runtime-minimal-ubi9-python-3-9-pr-image-mirror
/override multi-pr-opendatahub-io-notebooks-792-openshift-release-59520-runtime-minimal-ubi9-python-3-11-pr-image-mirror

these multi-pr checks don't work for testing build configs updates in openshift/release

**Comment by jiridanek** [comment]
/override "build / (cuda-jupyter-tensorflow-ubi9-python-3.11)"

**Comment by jiridanek** [comment]
Can't ever get all images to build, so OCP-CI tests don't run

> 

: Build image rocm-ubi9-python-3.11 from the repository expand_more42m5s | : Build image rocm-ubi9-python-3.11 from the repository expand_more | 42m5s
-- | -- | --
: Build image rocm-ubi9-python-3.11 from the repository expand_more | 42m5s
: Build image runtime-cuda-tensorflow-ubi9-python-3.11 from the repository expand_more | : Build image runtime-cuda-tensorflow-ubi9-python-3.11 from the repository expand_more
: Build image runtime-cuda-tensorflow-ubi9-python-3.11 from the repository expand_more

/override ci/prow/rocm-runtimes-ubi9-e2e-tests ci/prow/rocm-notebooks-e2e-tests ci/prow/notebooks-ubi9-e2e-tests 
/override ci/prow/rstudio-notebook-e2e-tests ci/prow/rocm-runtimes-ubi9-e2e-tests ci/prow/codeserver-notebook-e2e-tests

**Comment by jiridanek** [comment]
/override ci/prow/intel-notebooks-e2e-tests ci/prow/runtimes-ubi9-e2e-tests

**Pull request merged**